designer/client/src/components/aiAssistant/AiAssistantProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import type { ChatModelRunOptions, TextContentPart } from "@assistant-ui/react";
+import { AssistantRuntimeProvider, useLocalRuntime, type ChatModelAdapter } from "@assistant-ui/react";
+import { createParser } from "eventsource-parser";
+import type { ReactNode } from "react";
+import React from "react";
+
+import httpService from "../../http/HttpService";
+
+async function initializeChatStream(
+    messages: ChatModelRunOptions["messages"],
+    abortSignal?: AbortSignal,
+): Promise<{
+    responseParts: string[];
+    state: { isAborted: boolean; isFinished: boolean };
+}> {
+    const response = await httpService.sendChatMessage(messages[messages.length - 1].content[0] as TextContentPart, abortSignal);
+    const responseParts: string[] = [];
+    const state = { isAborted: false, isFinished: false };
+
+    const parser = createParser({
+        onEvent({ event, data }) {
+            if (event === "delta") {
+                responseParts.push(JSON.parse(data).text);
+            }
+            if (event === "stop") {
+                state.isFinished = true;
```

Comment:
Would it be possible to remember also generated `threadId` by BE so that FE could pass the id in subsequent messages from user?

Reply:
I added a little hacky approach here to remember threadId. It looks like the library generates threadId on the front end, and I haven't found a way to override it yet. There are some GitHub discussions about it https://github.com/assistant-ui/assistant-ui/issues/1780 but there is no conclusion yet.